src/microscope_focus_map/focus_map.py

```diff
@@ -0,0 +1,150 @@
+"""
+Microscope Focus Map Algorithm.
+
+Usage:
+    focus_map.py bilinear [<points>]
+    focus_map.py biquadratic [<points>]
+    focus_map.py bicubic [<points>]
+    focus_map.py poly <order> [<points>]
+    focus_map.py -h | --help
+
+Options:
+    -h --help           Show help screen.
+
+For more information see README.md
+"""
+
+# Code adapted from:
+# https://math.stackexchange.com/questions/99299/best-fitting-plane-given-a-set-of-points
```

Comment:
Probably should link to the answer you're using, there are a few there and it's not totally clear where to start otherwise: https://math.stackexchange.com/questions/99299/best-fitting-plane-given-a-set-of-points#answer-2306029

Reply:
Done